src/renderer/components/message/MessageWrapper.tsx

```diff
@@ -18,30 +28,28 @@ export const render = props => {
 }
 
 export const RenderMessage = React.memo(
-  props => {
+  (props: RenderMessageProps) => {
     const { message, locationStreamingEnabled, chat, chatStoreDispatch } = props
     const { fromId, id } = message.msg
     const msg = message.msg
     const tx = window.translate
     const screenContext = useContext(ScreenContext)
     const { openDialog } = screenContext
 
-    message.onForward = () => openDialog('ForwardMessage', { message })
-
-    const conversationType = GROUP_TYPES.includes(chat.type)
+    const conversationType: 'group' | 'direct' = GROUP_TYPES.includes(chat.type)
       ? 'group'
       : 'direct'
```

Comment:
This is one of the things i dont like about typescript, you have the types there twice and it's getting really verbose. I sometimes have the feeling that I need to duplicate type annotations quite a lot.

Reply:
the typescript compile is not the smartest... But if it were, build times would probably be longer.